### DISCUSSION TITLE
What changed from Refit 6.3.2 to 7.0?

### DISCUSSION THREAD

**Discussion by dansiegel** [reactiveui/refit#1621] [General]
Ok silly question that is oddly more challenging to answer than I thought it would be. What changed from Refit 6.x to Refit 7.0? It's been out for 5 months, but I don't see any public GitHub release or tag for it and the Readme is quick to talk about changes in 6.x but doesn't say anything about 7. I would assume for a major version change something more significant must have changed but I cannot answer what that is currently.

**Comment by anaisbetts** [comment]
There was a change in 6.5 that we ended up reworking, https://github.com/reactiveui/refit/pull/1495 has the details. Since it changed a public API it ended up being a major version change, though likely the vast majority of people don't even notice

**Reply by hoyleg** [reply]
Was 6.5.1 never regarded as publishable?  I would like to use the RestMethodInfo but v7 creates a conflict between library versions as my entire codebase is .Net v6.

**Reply by anaisbetts** [reply]
Refit v7 has nothing to do with .NET v7, the numbers are solely coincidence

**Comment by glennawatson** [comment]
Summary for anyone landing here:

- Refit 7.0 is a major bump only because a public API surface changed (RestMethodInfo and related), reworking what shipped in 6.5. Details in PR #1495.
- It is not tied to .NET 7. The "7" is coincidental; Refit 7.x still targets older runtimes including .NET 6 (it multi-targets netstandard2.0 and several net TFMs).
- For most consumers it is effectively a drop-in upgrade; the break only affects code that referenced the changed public types directly.

If you need the older RestMethodInfo shape specifically and cannot adapt, pin to 6.x; otherwise 7.x works fine on a .NET 6 codebase.
